### PR TITLE
Add header to SCC endpoint

### PR DESCRIPTION
**Pull request by jesusbv**
## Description

Add test to check the header is sent

## Change Type

*Please select the correct option.*

- [X] **Bug Fix** (a non-breaking change which fixes an issue)
- [ ] **New Feature** (a non-breaking change which adds new functionality)
- [ ] **Documentation Update** (a change which only updates documentation)

## Checklist

*Please check off each item if the requirement is met.*

- [X] I have reviewed my own code and believe that it's ready for an external review.
- [X] I have provided comments for any hard-to-understand code.
- [ ] I have documented the `MANUAL.md` file with any changes to the user experience.
- [ ] If my changes are non-trivial, I have added a changelog entry to notify users at `package/obs/rmt-server.changes`.

## Review

Please check out our [review guidelines](https://github.com/SUSE/scc-docs/blob/master/team/workflow/code_review.md) 
and get in touch with the author to get a shared understanding of the change.

### CODE REVIEW

**Review by bear454 — APPROVED** [review]

**Review by rjschwei — APPROVED** [review]

**Comment by rjschwei** [comment]
@digitaltom @felixsch if this looks good to you please merge and create a RMT 2.19 package for us to test.

**Comment by jesusbv** [comment]
> @digitaltom @felixsch if this looks good to you please merge and create a RMT 2.19 package for us to test.

I understand the testing package should include #1204 (hopefully, once merged) as well

File: engines/registry/app/models/access_scope.rb [hunk 1]
```diff
@@ -91,8 +91,11 @@ def allowed_paths(system = nil)
     if system && system.hybrid?
       allowed_product_class = allowed_products.detect { |s| s.downcase.include?('ltss') }
```
Comment (digitaltom) [inline engines/registry/app/models/access_scope.rb:92]: It looks weird to hardcode 'ltss' here. Why is this not compared with the actual product class?

Comment (jesusbv) [reply engines/registry/app/models/access_scope.rb:92]: I want to check if the product needs to check subscription with SCC
the product class is, i.e., `SLES15-SP4-LTSS-X86`  if Im not mistaken
so, checking for `ltss` to include any version or arch made sense to me

Comment (jesusbv) [reply engines/registry/app/models/access_scope.rb]: using a check for non free product

File: engines/registry/app/models/access_scope.rb [hunk 2]
```diff
@@ -88,19 +88,19 @@ def allowed_paths(system = nil)
     allowed_products = (active_products & access_policies_yml.keys)
```
Comment (digitaltom) [inline engines/registry/app/models/access_scope.rb:88]:
```suggestion
    allowed_product_classes = (active_products & access_policies_yml.keys)
```

also above, it should read `active_product_classes`

File: engines/registry/app/models/access_scope.rb [hunk 3]
```diff
@@ -88,19 +88,19 @@ def allowed_paths(system = nil)
...
     allowed_glob_paths = access_policies_yml.values_at(*allowed_products).flatten
 
-    if system && system.hybrid?
-      allowed_product_class = allowed_products.detect { |s| s.downcase.include?('ltss') }
-      if allowed_product_class.present?
-        activation_state = SccProxy.scc_check_subscription_expiration(
-          {}, system.login, system.system_token, Rails.logger, system.proxy_byos_mode, allowed_product_class
+    if system && system.hybrid? && allowed_products.any? { |s| !Product.find_by(product_class: s).free? }.present?
```
Comment (digitaltom) [inline engines/registry/app/models/access_scope.rb:91]: We'll need a comment here what this condition is doing. I guess it's something like `# if the system is hybrid, check if it's subscription is still valid for accessing images`
Why is this only checking for hybrid systems and not BYOS?

File: engines/registry/app/models/access_scope.rb [hunk 4]
```diff
@@ -88,19 +88,19 @@ def allowed_paths(system = nil)
...
+      auth_header = {
+        Authorization: ActionController::HttpAuthentication::Basic.encode_credentials(system.login, system.password)
+      }
+      activation_state = SccProxy.scc_check_subscription_expiration(
+        auth_header, system.login, system.system_token, Rails.logger, system.proxy_byos_mode, allowed_products.first
```
Comment (digitaltom) [inline engines/registry/app/models/access_scope.rb]: It feels random to use the `.first` product here. Are we sure that the first product is actually the product that was activated with SCC?

Comment (jesusbv) [reply engines/registry/app/models/access_scope.rb]: we have a product class, one product class

in `allowed_product_classes` (which should be `allowed_product_class` will be always one product class, correct ?
like access policy has `- FOO: /a/b/c/**` and the paths can be 18 but `FOO` is one, yes it comes in an array but it is an array of size 1

Comment (jesusbv) [reply engines/registry/app/models/access_scope.rb]: Update the approach, to remove the non active non free extensions from the allowed paths

**Comment by rjschwei** [comment]
> I cannot test run this, but code-wise looks ok now. Please also check how often that gets called and the performance of it, because the inline call to SCC can be expensive. Didn't we have a cache for this?

No there is no cache yet for checking if a system is registered and valid with SCC. That is a follow on project we can start once 2.19 is out.

**Pull request merged**